dockerfiles/hpcc/templates/_helpers.tpl

```diff
@@ -47,9 +47,25 @@ Generate global ConfigMap info
 Pass in root as .
 */}}
 {{- define "hpcc.generateGlobalConfigMap" -}}
+{{- $local := dict "defaultEsp" "" -}}
 imageVersion: {{ required "Please specify .global.image.version" .Values.global.image.version | quote }}
 singleNode: {{ .Values.global.singleNode | default false }}
-defaultEsp: {{ .Values.global.defaultEsp | default (index .Values.esp 0).name }}
+
+{{-/* Set the globalEsp to whatever was set globally, or the first non-disabled esp (if there is one) */-}}
+
+{{- if .Values.global.defaultEsp -}}
+ {{- $_ := set $local "defaultEsp" .Values.global.defaultEsp -}}
+{{- else if hasPrefix "[]" (typeOf .Values.esp) -}}
+ {{- range $key, $value := .Values.esp -}}
+  {{- if (not $value.disabled) -}}
+   {{- if (not $local.defaultEsp) -}}
+    {{- $_ := set $local "defaultEsp" $value.name -}}
+   {{- end -}} 
+  {{- end -}} 
+ {{- end -}} 
+{{- end }} 
+defaultEsp: {{ $local.defaultEsp | quote }}
```

Comment:
should this be conditionally generated?, i.e. inside a
```
{{- if $local.defaultEsp }}
```

Reply:
It's ok to set it to blank if not set